### PR TITLE
refactor: log statement on error by Display

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

This is a follow-up to #3646 and thus #3744 and #3792.

This can be related to #2144.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/frontend/src/instance/prom_store.rs [hunk 1]
```diff
@@ -149,9 +149,7 @@ impl Instance {
                 .handle_remote_query(&ctx, catalog_name, schema_name, &table_name, query)
                 .await
                 .map_err(BoxedError::new)
-                .with_context(|_| error::ExecuteQuerySnafu {
-                    query: format!("{query:#?}"),
-                })?;
+                .with_context(|_| error::ExecuteQuerySnafu)?;
```
Comment (tisonkun) [inline src/frontend/src/instance/prom_store.rs:152]: I test locally a sensitive query: `COPY tbl FROM '/path/to/folder/xxx.parquet' WITH (FORMAT = 'parquet', SECRET_ACCESS_KEY=12345);`

It logs:

```
2024-04-25T19:19:37.144010Z ERROR on_query{protocol="mysql"}:mysql::do_query:do_query: frontend::instance: Failed to execute query: COPY tbl FROM /path/to/folder/xxx.parquet WITH (format = 'parquet', secret_access_key = '******') err=0: Table operation error, at src/frontend/src/instance.rs:273:14
1: Failed to build data source backend, at /Users/tison/GreptimeWorkspace/greptimedb/src/operator/src/statement/copy_table_from.rs:98:59
2: Failed to build backend, at src/common/datasource/src/object_store/fs.rs:25:10
3: Unexpected (permanent) at Builder::build => create root dir failed

2024-04-25T19:19:37.144742Z ERROR on_query{protocol="mysql"}: servers::mysql::writer: Failed to handle mysql query err=0: Failed to execute query, query: COPY tbl FROM '/path/to/folder/xxx.parquet' WITH (FORMAT = 'parquet', SECRET_ACCESS_KEY=12345), at /Users/tison/GreptimeWorkspace/greptimedb/src/servers/src/query_handler/sql.rs:89:22
1: Table operation error, at src/frontend/src/instance.rs:273:14
2: Failed to build data source backend, at /Users/tison/GreptimeWorkspace/greptimedb/src/operator/src/statement/copy_table_from.rs:98:59
3: Failed to build backend, at src/common/datasource/src/object_store/fs.rs:25:10
4: Unexpected (permanent) at Builder::build => create root dir failed
```

As you can see, the Display implementation can be redacted properly. But `ExecuteQuerySnafu` leaks the query.

**Review by MichaelScofield — APPROVED** [review]

File: src/frontend/src/instance.rs [hunk 2]
```diff
@@ -448,9 +446,7 @@ impl PrometheusHandler for Instance {
             .execute_stmt(stmt, query_ctx.clone())
             .await
             .map_err(BoxedError::new)
-            .with_context(|_| ExecuteQuerySnafu {
-                query: format!("{query:?}"),
-            })?;
+            .with_context(|_| ExecuteQuerySnafu)?;
```
Comment (evenyag) [inline src/frontend/src/instance.rs:449]:
```suggestion
            .context(ExecuteQuerySnafu)?;
```

File: src/frontend/src/instance/prom_store.rs (hunk 1 above)
Comment (evenyag) [inline src/frontend/src/instance/prom_store.rs:152]:
```suggestion
                .context(error::ExecuteQuerySnafu)?;
```

File: src/servers/src/query_handler/sql.rs [hunk 3]
```diff
@@ -109,12 +106,8 @@ where
...
             .into_iter()
             .map(|x| {
-                x.map_err(BoxedError::new).with_context(|_| {
-                    let query_literal = format!("{query:?}");
-                    error::ExecuteQuerySnafu {
-                        query: query_literal,
-                    }
-                })
+                x.map_err(BoxedError::new)
+                    .with_context(|_| error::ExecuteQuerySnafu)
```
Comment (evenyag) [inline src/servers/src/query_handler/sql.rs:110]:
```suggestion
                    .context(error::ExecuteQuerySnafu)
```

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**